### PR TITLE
Add BUILD_EXE as cmake option

### PR DESCRIPTION
**Pull request by bam241**
This should allow to build dagmc libs (shared and.or static) without the exe.

### CODE REVIEW

File: cmake/DAGMC_macros.cmake [hunk 1]
```diff
@@ -259,64 +264,72 @@ endmacro ()
...
     endif ()
+    install(TARGETS ${exe_name} DESTINATION ${INSTALL_BIN_DIR})
   endif ()
-  install(TARGETS ${exe_name} DESTINATION ${INSTALL_BIN_DIR})
 endmacro ()
 
 # Install a unit test
 macro (dagmc_install_test test_name ext)
-  message(STATUS "Building unit tests: ${test_name}")
+  if (NOT BUILD_EXEC)
```
Comment (ljacobson64) [inline cmake/DAGMC_macros.cmake]: This should be `BUILD_EXE` not `BUILD_EXEC`

**Comment by bam241** [comment]
I am now missing the includes :( in the install folder....

**Comment by ljacobson64** [comment]
I think that's a problem in the `dagmc_install_library` macro, not something you did. The shared library section starts like this:

```
  if (BUILD_SHARED_LIBS)
    add_library(${lib_name}-shared SHARED ${SRC_FILES})
      set_target_properties(${lib_name}-shared
        PROPERTIES OUTPUT_NAME ${lib_name}
        PUBLIC_HEADER "${PUB_HEADERS}")
```

while the static library section starts like this:

```
  if (BUILD_STATIC_LIBS)
    add_library(${lib_name}-static STATIC ${SRC_FILES})
    set_target_properties(${lib_name}-static
      PROPERTIES OUTPUT_NAME ${lib_name})
```

The static library section doesn't set the public headers as part of the target properties.

**Comment by bam241** [comment]
included the static header install fix in this PR....

File: cmake/DAGMC_macros.cmake [hunk 2]
```diff
@@ -66,6 +66,7 @@ macro (dagmc_setup_options)
   option(BUILD_SHARED_LIBS "Build shared libraries" ON)
   option(BUILD_STATIC_LIBS "Build static libraries" ON)
 
+  option(BUILD_EXE "Build DAGMC executables" ON)
```
Comment (pshriwise) [inline cmake/DAGMC_macros.cmake]: Let's indent the comment string to match the other options.

File: cmake/DAGMC_macros.cmake [hunk 3]
```diff
@@ -96,12 +97,14 @@ endif()
...
   endif ()
-  if (NOT BUILD_SHARED_LIBS AND NOT BUILD_STATIC_EXE)
-    message(FATAL_ERROR "BUILD_STATIC_EXE cannot be OFF while BUILD_SHARED_LIBS is OFF")
-  endif ()
   if (NOT BUILD_SHARED_LIBS AND NOT BUILD_STATIC_LIBS)
     message(FATAL_ERROR "BUILD_SHARED_LIBS and BUILD_STATIC_LIBS cannot both be OFF")
   endif ()
+  if (NOT BUILD_SHARED_LIBS AND NOT BUILD_STATIC_EXE AND BUILD_EXE)
+    SET(BUILD_EXE OFF)
+    message("Turning BUILD_EXE to OFF: SHARED_BUIL_LIBS and BUILD_STATIC_EXE are OFF")
```
Comment (pshriwise) [inline cmake/DAGMC_macros.cmake]:
```suggestion
    message("Turning BUILD_EXE OFF: SHARED_BUILD_LIBS and BUILD_STATIC_EXE are OFF")
```

File: news/PR-0717.rst [hunk 4]
```diff
@@ -0,0 +1,13 @@
+**Added:**
+- adding BUILD_EXE option (default ON) alowing to build only the dagmc libs
+  without the executable (for static and/or shared libs)
+
+**Changed:** None
+- now install dagmc header when building and instaling static libs
```
Comment (pshriwise) [inline news/PR-0717.rst]: Some formatting updates so the news file renders correctly.
```suggestion
**Added:**

- adding BUILD_EXE option (default ON) alowing to build only the dagmc libs
  without the executable (for static and/or shared libs)

**Changed:** None

- now install dagmc header when building and instaling static libs

```

File: cmake/DAGMC_macros.cmake (hunk 3 above)
Comment (pshriwise) [reply cmake/DAGMC_macros.cmake]: I'll suggest we make this a warning message so it's more noticeable.

**Review by pshriwise — CHANGES_REQUESTED** [review]
Just a few small items from me here.

**Comment by bam241** [comment]
@pshriwise is there some still some undressed comments in here ?

**Comment by ljacobson64** [comment]
@bam241 can you remind me what the use case is for building without executables?

**Comment by bam241** [comment]
> @bam241 can you remind me what the use case is for building without executables?

The plugin.

**Review by pshriwise — APPROVED** [review]
All my comments were addressed. Looks good @bam241!

**Review by gonuke — APPROVED** [review]
LGTM - thanks @bam241

**Pull request merged**